### PR TITLE
Add IvoryCKEditorBundle integration for admin

### PR DESCRIPTION
**Pull request by WouterJ**
Fixes https://github.com/symfony-cmf/SimpleCmsBundle/issues/132

### CODE REVIEW

File: Admin/PageAdmin.php [hunk 1]
```diff
@@ -45,6 +51,7 @@ protected function configureListFields(ListMapper $listMapper)
     protected function configureFormFields(FormMapper $formMapper)
     {
         parent::configureFormFields($formMapper);
+        $ivoryCKeditorAvailable = class_exists('Ivory\CKEditorBundle\IvoryCKEditorBundle') && $this->ivoryCkeditor;
```
Comment (lsmith77) [inline Admin/PageAdmin.php]: why not do this in the DI extension? ie. if a config is set, check if the Bundle is available .. if not throw an error.

Comment (WouterJ) [inline Admin/PageAdmin.php]: we should not thrown an exception: we just fallback to textarea if the bundle is not available.

I would like to have this even more configurable (by allowing to configure field type and field options, both with a default based on the available bundles). But I'm not sure if we're over doing the configuration side for a bundle in such cases.

In 2.0, we want to extra all form stuff from the admin (so we can use it for Kunstmaan as well). We can make such things customizable by extending the form type in such cases.

Comment (lsmith77) [inline Admin/PageAdmin.php]: but if someone configures it .. why should be degrade silently?

Comment (lsmith77) [inline Admin/PageAdmin.php]: but if you prefer silent degration .. you can still check in the DI extension and then simply choose to ignore the config if the Bundle isn't installed

Comment (ElectricMaxxx) [inline Admin/PageAdmin.php]: I would also prefer a solution, where the existence check is done in the extension 

Options set + bundle there => fine 
Options set + bundle not there => exceprion

I think WE did some equal with the burgov-bundle in CmfSeo.

**Comment by ElectricMaxxx** [comment]
Btw, this will need a nice documentation to highlight this feature as it brings so much value by doing almost nothing.

**Comment by WouterJ** [comment]
Updated the PR with better error reporting and configuration.

File: DependencyInjection/CmfSimpleCmsExtension.php [hunk 2]
```diff
@@ -70,6 +70,36 @@ public function load(array $configs, ContainerBuilder $container)
...
+
+            if (class_exists('Ivory\CKEditorBundle\IvoryCKEditorBundle')) {
+                $message .= ' (did you forget to register the bundle in the AppKernel?)';
+            }
+
+            throw new \LogicException($message.'.');
+        }
+
+        if (false === $config['enabled'] || !isset($bundles['IvoryCKEditorBundle'])) {
+            $container->setParameter($this->getAlias() . '.ivory_ckeditor.config', array());
```
Comment (lsmith77) [inline DependencyInjection/CmfSimpleCmsExtension.php]: you need to configure a default for this parameter:
```
 [Symfony\Component\DependencyInjection\Exception\ParameterNotFoundException]  
  The service "cmf_simple_cms.persistence.phpcr.admin.page" has a dependency    
  on a non-existent parameter "cmf_simple_cms.ivory_ckeditor.config". 
```

Comment (WouterJ) [inline DependencyInjection/CmfSimpleCmsExtension.php]: fixed

**Pull request merged**